### PR TITLE
vampire: update to 2024-04-12

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Long-due update

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by ryandesign** [comment]
I did not see this PR before committing 7906951325b04e741367706b275524afec74d311 and filing https://trac.macports.org/ticket/69761 but shouldn't the port be updated to the latest stable version rather than to a random commit?

**Comment by barracuda156** [comment]
@ryandesign I have been communicating with upstream from December to fix this port: https://github.com/vprover/vampire/issues/512
Much stuff has been done.

So no, please, keep it commit-based until the next release. I am not going to backport fixes and re-test everything again.

File: math/vampire/Portfile [hunk 1]
```diff
@@ -1,24 +1,23 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           github 1.0
 PortGroup           cmake 1.1
+PortGroup           github 1.0
 
 # Until a new release is published under the 3-clause BSD license, we have to
 # track the development branch; as of this writing, the last stable
 # release (4.5.1) was made under the previous, restricted-use license.
```
Comment (ryandesign) [inline math/vampire/Portfile:9]: This comment no longer applies; you could remove this or rewrite it to explain the new reason why we're not using a release.

Comment (barracuda156) [reply math/vampire/Portfile]: Done

**Pull request merged**